executor/change.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+import (
+	"context"
+	"strings"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/parser/ast"
+	"github.com/pingcap/tidb-tools/tidb-binlog/node"
+	"github.com/pingcap/tidb/config"
+	"github.com/pingcap/tidb/util/chunk"
+)
+
+// ChangeExec represents a change executor.
+type ChangeExec struct {
+	baseExecutor
+
+	Statement *ast.ChangeStmt
+}
+
+// Next implements the Executor Next interface.
+func (e *ChangeExec) Next(ctx context.Context, req *chunk.RecordBatch) error {
+	stmt := e.Statement
+	return updateNodeState(ctx, stmt)
+}
+
+// updateNodeState update pump or drainer's state.
```

Comment:
Why not put the implementation of this function in ChangeExec.Next directly?

Reply:
In my option, this method may be merged into tidb-tools,  this can make merge easier.

Reply:
finished